ui/page/transaction/transactions_page.go

```diff
@@ -147,163 +146,271 @@ func (pg *TransactionsPage) pageTitle(gtx C) D {
 	return txt.Layout(gtx)
 }
 
+func (pg *TransactionsPage) getAssetType() utils.AssetType {
+	if pg.selectedWallet == nil {
+		return utils.DCRWalletAsset
+	}
+	return pg.selectedWallet.GetAssetType()
+}
+
 func (pg *TransactionsPage) refreshAvailableTxType() {
-	wal := pg.WL.SelectedWallet.Wallet
 	items := []cryptomaterial.DropDownItem{}
-	_, keysinfo := components.TxPageDropDownFields(wal.GetAssetType(), pg.selectedTabIndex)
+	_, keysinfo := components.TxPageDropDownFields(pg.getAssetType(), pg.selectedTxCategoryTab)
 	for _, name := range keysinfo {
-		item := cryptomaterial.DropDownItem{}
-		item.Text = name
-		items = append(items, item)
+		items = append(items, cryptomaterial.DropDownItem{Text: name})
 	}
-
 	pg.txTypeDropDown = pg.Theme.DropDown(items, values.TxDropdownGroup, 2)
 
-	go func() {
-		countfn := func(fType int32) int {
-			count, _ := wal.CountTransactions(fType)
-			return count
+	// only show tx count for regular txs, not staking
+	if pg.selectedTxCategoryTab == 0 {
+		pg.showLoader = true
+
+		wallets := pg.assetWallets
+		if pg.selectedWallet != nil {
+			wallets = []sharedW.Asset{pg.selectedWallet}
 		}
 
-		items := []cryptomaterial.DropDownItem{}
-		mapinfo, keysinfo := components.TxPageDropDownFields(wal.GetAssetType(), pg.selectedTabIndex)
-		for _, name := range keysinfo {
-			fieldtype := mapinfo[name]
-			item := cryptomaterial.DropDownItem{}
-			if pg.selectedTabIndex == 0 {
-				item.Text = fmt.Sprintf("%s (%d)", name, countfn(fieldtype))
-			} else {
-				item.Text = name
+		// Do this in background to prevent the app from freezing when counting
+		// wallet txs. This is needed in situations where the wallet has lots of
+		// txs to be counted.
+		go func() {
+			items := []cryptomaterial.DropDownItem{}
+			mapinfo, keysinfo := components.TxPageDropDownFields(pg.getAssetType(), pg.selectedTxCategoryTab)
+			for _, name := range keysinfo {
+				var txTypeCount int
+				for _, wal := range wallets {
+					count, _ := wal.CountTransactions(mapinfo[name])
+					txTypeCount += count
+				}
+				items = append(items, cryptomaterial.DropDownItem{
+					Text: fmt.Sprintf("%s (%d)", name, txTypeCount),
+				})
 			}
-			items = append(items, item)
+
+			pg.txTypeDropDown = pg.Theme.DropDown(items, values.TxDropdownGroup, 2)
+			pg.showLoader = false
+			pg.ParentWindow().Reload()
+		}()
+	}
+}
+
+func (pg *TransactionsPage) fetchTransactions(offset, pageSize int32) (txs []*multiWalletTx, totalTxs int, isReset bool, err error) {
+	wal := pg.selectedWallet
+	if wal == nil {
+		txs, totalTxs, err = pg.multiWalletTxns(offset, pageSize)
+	} else {
+		txs, totalTxs, err = pg.loadTransactions(wal, offset, pageSize)
+	}
+
+	isReset = pg.previousTxFilter != pg.txFilter
+	if isReset {
+		// reset the offset to zero
+		offset = 0
```

Comment:
This is probably where you lint error is coming from, why is offset set to zero here, it servers no purpose